app/main/util/render_utils.py

```diff
@@ -0,0 +1,220 @@
+    elif file_type in ["png", "jpg", "jpeg"]:
+        return f"image/{file_type}"
+
+
+def get_file_details(file):
+    """Retrieve file metadata and determine file type and extension."""
+    file_metadata = get_file_metadata(file.FileId)
+    file_extension = file.FileName.split(".")[-1].lower()
+
+    if file_extension in ["pdf", "png", "jpg", "jpeg"]:
+        file_type = "iiif"
+    else:
+        file_type = None
+
+    return file_metadata, file_type, file_extension
+
+
+def generate_breadcrumb_values(file):
+    """Generate breadcrumb values for the record template."""
+    consignment = file.consignment
+    body = consignment.series.body
+    series = consignment.series
+    return {
+        0: {"transferring_body_id": body.BodyId},
+        1: {"transferring_body": body.Name},
+        2: {"series_id": series.SeriesId},
+        3: {"series": series.Name},
+        4: {"consignment_id": consignment.ConsignmentId},
+        5: {"consignment_reference": consignment.ConsignmentReference},
+        6: {"file_name": file.FileName},
+    }
+
+
+def get_download_filename(file):
+    """Generate download filename for a file."""
+    if file.CiteableReference:
+        if len(file.FileName.rsplit(".", 1)) > 1:
+            return (
+                file.CiteableReference + "." + file.FileName.rsplit(".", 1)[1]
+            )
+    return None
+
+
+def manage_static_file(file, file_extension):
+    """Manage the file in the static directory."""
+    s3 = boto3.client("s3")
+    bucket = current_app.config["RECORD_BUCKET_NAME"]
+    key = f"{file.consignment.ConsignmentReference}/{file.FileId}"
+    files_directory = os.path.join(current_app.static_folder, "files")
+    static_file_path = ""
+
+    if file_extension == "pdf":
+        static_file_path = os.path.join(files_directory, "temp_file.pdf")
+    elif file_extension in ["png", "jpg", "jpeg"]:
+        static_file_path = os.path.join(
+            files_directory, f"temp_file.{file_extension}"
+        )
+    else:
+        static_file_path = os.path.join(
+            files_directory, f"temp_file.{file_extension}"
+        )
+
+    if os.path.exists(files_directory):
+        shutil.rmtree(files_directory)
+
+    os.makedirs(files_directory)
+
+    s3_file_object = s3.get_object(Bucket=bucket, Key=key)
+    file_content = s3_file_object["Body"].read()
+
+    with open(static_file_path, "wb") as static_file:
+        static_file.write(file_content)
+
+    return static_file_path
+
+
+def generate_pdf_manifest(record_id):
+    file = db.session.get(File, record_id)
+
+    if file is None:
+        abort(404)
+
+    file_name = file.FileName
+    file_url = url_for(
+        "main.download_record", record_id=record_id, _external=True
+    )
+
+    manifest = {
+        "@context": [
+            "http://iiif.io/api/presentation/3/context.json",
+        ],
+        "id": f"{url_for('main.download_record', record_id=record_id, _external=True, render=True)}",
```

Comment:
I just realised that id should be the url of the manifest online, not the url of the file. https://iiif.io/api/cookbook/recipe/0001-mvm-image/

`The id property identifies this manifest with the URL at which it is available online.`